libgobuster/statehelpers.go

```diff
@@ -170,6 +171,22 @@ func ValidateDirModeState(
 		}
 	}
 
+	// Read the user agents file
+	if errorList.ErrorOrNil() == nil && s.RandomUserAgent {
+		b, err := ioutil.ReadFile(s.UserAgentsFile)
```

Comment:
`ioutil.ReadFile` will read the entire file into memory, then you split on newlines, remove comments/etc. It may not matter super much here, but there is a potential that we read in a lot of stuff that is never required, etc.

Following the pattern used for reading the , we could use `bufio.NewScanner` to read/process the file line by line:

* https://github.com/OJ/gobuster/blob/master/libgobuster/state.go#L143
* https://github.com/OJ/gobuster/blob/master/libgobuster/state.go#L158-L168

So an implementation in this case might look something like:

```
for scanner.Scan() {
		if s.Terminate {
			break
		}
		line := strings.TrimSpace(scanner.Text())

		// Skip "comment" (starts with #), as well as empty lines
		if !strings.HasPrefix(word, "#") && len(line) > 0 {
			uas = append(uas, line)
		}
	}
```

If we wanted to be fancy, we could probably refactor the common functionality between them into a helper function.. But that is probably beyond what is required here.